src/devices/sound/mas3507d.h

```diff
@@ -25,7 +25,13 @@ class mas3507d_device : public device_t, public device_sound_interface
 	void reset_sample_count() { total_sample_count = 0; }
 	u32 get_sample_count() const { return total_sample_count; }
 
-	void set_playback_enabled(bool enabled) { playback_enabled = enabled; }
+	void set_playback_enabled(bool enabled) {
+		playback_enabled = enabled;
+		reset_sample_count();
```

Comment:
The game wants the sample counter to be 0 when audio is stopped so it knows playback has ended, and logically it should be 0 when the audio starts. So it's easiest to throw it in here so it's guaranteed to be reset when the playback state changes.

Reply:
Are you sure it's a sample counter?  It felt like a mpeg frame counter when I studied the fpga.

Reply:
I was just going with the terminology you used in your rewrite. I feel like you're probably right that it's a frame counter since the mas3507d never deals with "samples" from what I've read in the datasheets, but it definitely does return the current frame count.

```
The counter will be incremented with each new frame
that is decoded. With an invalid MPEG bit stream as its
input (e.g. if an invalid header is detected), the
MAS 3507D resets the MPEGFrameCount cell to ‘0’.
The MPEGFrameCount is also returned by the ‘default
read’ command as described in Section 3.3.9.
```

I'll see about following this more strictly in the code (in particular resetting on a bad frame, since I don't think that's current done at all?).

Reply:
I've rewritten this code. I tried making it increment by 1 for every new frame but that doesn't seem to be it. I also tried incrementing by a fixed number like 1000 but that's not it either. Drummania relies on this feature to work properly, and if it's not incremented properly then the game stops moving the chart field entirely.

Incrementing by the number of samples decoded gives perfect timing, so I think that's probably right. I also know from my homebrew tool thing that it's definitely not incrementing by 1 because the value increases really quickly but I haven't actually tried to determine what the step value was because it goes up too quickly.

Checked with DDR Extreme, Drummania 9th Mix, and DDR Solo Bass Mix.